src/app/core/basic-datatypes/number/number-range-filter/range-input/range-input.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, ElementRef, Input, Optional, Self } from "@angular/core";
+import {
+  FormControl,
+  FormControlDirective,
+  FormGroup,
+  FormGroupDirective,
+  NgControl,
+  NgForm,
+  ReactiveFormsModule,
+  ValidationErrors,
+  ValidatorFn,
+} from "@angular/forms";
+import { ErrorStateMatcher } from "@angular/material/core";
+import { MatFormFieldControl } from "@angular/material/form-field";
+import { CustomFormControlDirective } from "app/core/common-components/basic-autocomplete/custom-form-control.directive";
+import { MatInput } from "@angular/material/input";
+
+@Component({
+  selector: "app-range-input",
+  standalone: true,
+  imports: [MatInput, ReactiveFormsModule],
+  templateUrl: "./range-input.component.html",
+  styleUrl: "./range-input.component.scss",
+  providers: [
+    { provide: MatFormFieldControl, useExisting: RangeInputComponent },
+  ],
+})
+export class RangeInputComponent extends CustomFormControlDirective<NumericRange> {
+  formGroup: FormGroup = new FormGroup({
+    from: new FormControl(),
+    to: new FormControl(),
+  });
+  formControlDirective: FormControlDirective;
+
+  @Input() override set value(value: NumericRange) {
+    // update the internal formGroup when the value changes from the outside
+    this.formGroup.setValue(value, { emitEvent: false });
+    super.value = value;
+  }
+  override get value(): NumericRange {
+    return super.value;
+  }
+
+  @Input()
+  activateValidation: boolean = true;
```

Comment:
```suggestion
  /**
   * Validation (activated by default) ensures the component has a sensible range
   * (e.g. "from" <= "to")
   * or otherwise marks the formControl invalid
   */
  @Input() activateValidation: boolean = true;
```